pkg/controller/provider/web/graph/graphql.go

```diff
@@ -0,0 +1,52 @@
+package graph
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/konveyor/controller/pkg/logging"
+	"github.com/konveyor/forklift-controller/pkg/controller/provider/web/base"
+	"github.com/konveyor/forklift-controller/pkg/controller/provider/web/graph/generated"
+
+	"github.com/99designs/gqlgen/graphql/handler"
+	"github.com/99designs/gqlgen/graphql/playground"
+)
+
+var log = logging.WithName("web|graphql")
+
+//
+// Routes.
+const (
+	GraphqlRoot = Root
+)
+
+//
+// GraphQL handler.
+type GraphHandler struct {
+	base.Handler
+}
+
+//
+// Add routes to the `gin` router.
+func (h *GraphHandler) AddRoutes(e *gin.Engine) {
+	e.POST(GraphqlRoot, h.GraphqlHandler())
+	e.GET(GraphqlRoot+"/playground", playgroundHandler())
+}
+
```

Comment:
To be consistent with other handlers, recommend something like:
```
e.POST(GraphqlRoot, h.Post)
e.GET(GraphqlRoot+"/playground", h.Get)

func (h GraphHandler) Post(ctx *gin.Context) {
    err := r.updateContext(ctx)
    if err != nil {
        ctx.Status(http.StatusInternalServerError)
        return
    }

    r.qqlh.ServeHTTP(ctx.Writer, ctx.Request)
}

func (h GraphHandler) Get(ctx *gin.Context) {
    err := r.updateContext(ctx)
    if err != nil {
        ctx.Status(http.StatusInternalServerError)
        return
    }

    r.playground.ServeHTTP(ctx.Writer, ctx.Request)
}

func (h *GraphHandler) updateContext(ctx *gin.Context) (err error) {
    # inject container into the ctx.Request.Context
    return
}
```

The GraphHandler.gqlh and GraphHandler.playground can be assigned when the Handler is constructed in doc.go.

Reply:
Makes sense.